src/object/cli_obj.c

```diff
@@ -3788,11 +3788,10 @@ obj_comp_cb(tse_task_t *task, void *data)
 		 * are some other cases we need to retry the RPC with current
 		 * shard, such as -DER_TIMEDOUT or daos_crt_network_error().
 		 */
-
-		if ((!obj_auxi->spec_shard && !obj_auxi->spec_group &&
-		     !obj_auxi->no_retry) ||
-		    (task->dt_result != -DER_INPROGRESS &&
-		     task->dt_result != -DER_TX_BUSY))
+		/* It needs to retry for INPROGRESS and TX_BUSY anyway */
+		if (!obj_auxi->no_retry &&
+		    (!obj_auxi->spec_shard || task->dt_result == -DER_INPROGRESS ||
+					      task->dt_result == -DER_TX_BUSY))
```

Comment:
this change forbidden retry for some cases of "spec_shard", but actually it need to retry for some case, for example just hit ESTALE/TIMEOUT/network_err.
This is the reason cause CI failures.
Seems this will not affect soak test, so I'll leave you to update it.

Reply:
I refreshed this PR (what i changed is in the 2nd commit), just to fix the CI failures and avoid affect soak test. (As this PR is merged to PR6595).
Please check, if my change is not good please feel free to drop/change, thanks.